abspath.c

```diff
@@ -5,7 +5,7 @@
  * symlink to a directory, we do not want to say it is a directory when
  * dealing with tracked content in the working tree.
  */
-int is_directory(const char *path)
+int dir_exists(const char *path)
```

Comment:
Looks good to me! Maybe the commit message could talk about the value of reconciling functionality, about consistency in naming (definitely mention that there is a `file_exists()` already), and it should probably talk about the differences of the `is_directory()` and the `dir_exists()` function, and that the former is a superset of the latter, but the latter has the better name.

Reply:
Sure. I'll work on it. Thank you.

Reply:
I already updated the commit message. Since I am not a native speaker, please feel free to correct me if you find the sentences weird.

Reply:
Looks good to me!

Reply:
(Please note that I am also not a native speaker...)